modules/react/_examples/stories/examples/MediaModal.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import {Modal} from '@workday/canvas-kit-react/modal';
+import {Box, BoxProps, Flex} from '@workday/canvas-kit-react/layout';
+import {Text} from '@workday/canvas-kit-react/text';
+import {PrimaryButton} from '@workday/canvas-kit-react/button';
+import {spaceNumbers} from '@workday/canvas-kit-react/tokens';
+
+const Image = (elemProps: React.ImgHTMLAttributes<HTMLImageElement>) => {
+  const [loaded, setLoaded] = React.useState(false);
+
+  return (
+    <Box
+      as="img"
+      display="block"
+      backgroundColor="frenchVanilla200"
+      transition="opacity ease 200ms"
+      opacity={loaded ? 1 : 0}
+      width="100%"
+      height={293}
+      onLoad={() => setLoaded(true)}
+      {...elemProps}
+    />
+  );
+};
+
+const MediaContainer = ({children, ...elemProps}: BoxProps) => {
+  return (
+    <Box
+      width={`calc(100% + ${spaceNumbers.l * 2}px)`}
+      marginLeft={`-${spaceNumbers.l}px`}
+      marginY="s"
+      {...elemProps}
+    >
+      {children}
+    </Box>
+  );
+};
```

Comment:
This is great, if you ever want to dive into creating your own compound component, I can see `Image` being a sub component of MediaContainer!

Reply:
Oh that's a brilliant idea! 

@dgubko you can read about [compound components](https://workday.github.io/canvas-kit/?path=/docs/guides-compound-components--page) and its benefits. Because `MediaContainer` will be a simple component without having a model (our component state management), you can use only `createComponent` [utility](https://workday.github.io/canvas-kit/?path=/docs/hooks-and-utilities-compound-components--create-component) to create a compound component structure for `MediaContainer` and `Image`.

Reply:
Thanks for the idea @mannycarrera4 and @RayRedGoose . I liked exploring compound components even though it was kinda hard!